app/_how-tos/kic-sticky-session-with-drain-support.md

```diff
@@ -0,0 +1,276 @@
+  a: Deploy {{ site.kic_product_name }} using the `--enable-drain-support=true` flag. Next, configure `spec.stickySessions` and set `spec.algorithm` to `sticky-sessions` in a `KongUpstreamPolicy` resource. Finally, attach the `KongUpstreamPolicy` resource to a Kubernetes Service with the `konghq.com/upstream-policy` annotation.
+
+prereqs:
+  kubernetes:
+    gateway_api: true
+  entities:
+    services:
+      - httpbin-service
+    routes:
+      - httpbin
+
+cleanup:
+  inline:
+    - title: Uninstall KIC from your cluster
+      include_content: cleanup/products/kic
+      icon_url: /assets/icons/kubernetes.svg
+
+---
+
+## Overview
+
+Sticky sessions ensure that repeat client requests are routed to the same backend pod, which is essential for maintaining user session state. When combined with drain support, you can implement graceful pod termination during deployments or scaling events, allowing existing sessions to complete while preventing new traffic from being routed to pods that are shutting down.
+
+This guide covers:
+1. How sticky sessions work in {{site.kic_product_name}}
+2. How to enable drain support for graceful terminations
+3. Configuring sticky sessions with a `KongUpstreamPolicy`
+4. Best practices for implementing these features together
+
+## Understanding Sticky Sessions
+
+Kong's sticky sessions feature uses browser-managed cookies to route repeat requests from the same client to the same backend target. When a client first connects, Kong sets a cookie in the response. On subsequent requests, if the cookie is present and valid, Kong routes the client to the same target.
+
+Sticky sessions are useful for:
+- Session persistence across multiple requests
+- Applications that store session state locally
+- Improving cache hit rates
+
+## Understanding Drain Support
+
+Drain support is a feature that allows Kong to gracefully handle terminating pods. When a pod begins the termination process in Kubernetes:
+
+1. The pod is marked for termination but continues running
+2. The pod's status changes to `Terminating`
+3. With drain support enabled, Kong:
+   - Identifies these terminating pods
+   - Adds them to the upstream with a weight of 0
+   - Allows existing connections to complete
+   - Prevents new connections from being routed to these pods
+
+This ensures a smooth transition during deployments, scaling events, or node maintenance.
+
+## Enabling Drain Support
+
+To enable drain support, you must start {{site.kic_product_name}} with the `--enable-drain-support` flag set to `true`. This can be done in your deployment YAML:
+
+```yaml
+containers:
+- name: ingress-controller
+  image: kong/kubernetes-ingress-controller:3.5
+  args:
+  - /kong-ingress-controller
+  - --enable-drain-support=true
+  # other arguments...
+```
+
+Alternatively, you can set the corresponding environment variable:
+
+```yaml
+env:
+- name: CONTROLLER_ENABLE_DRAIN_SUPPORT
+  value: "true"
+```
```

Comment:
I think we should we add a snippet how to set this in our helm chart. WDYT?

Reply:
Added. https://github.com/Kong/developer.konghq.com/pull/2131/commits/33f1c915b10093303504ca2118f2d4480e630c0f